include/rosdiscover-clang/BackwardSymbolizer/FunctionSymbolizer.h

```diff
@@ -718,6 +727,103 @@ class FunctionSymbolizer {
     return SymbolicFunctionCall::create(function, args);
   }
 
+
+  std::unique_ptr<SymbolicIfStmt> symbolizeIf(RawIfStatement* rawIf) {
+    clang::IfStmt *stmt = rawIf->getIfStmt();
+
+    llvm::outs() << "DEBUG: symbolizing if: ";
+    stmt->dump();
+    llvm::outs() << "\n";
+
+    auto value = boolSymbolizer.symbolize(stmt->getCond());
+    auto trueBranch = symbolizeCompound(rawIf->getTrueBody());
+    auto falseBranch = symbolizeCompound(rawIf->getFalseBody());
+
+    return std::make_unique<SymbolicIfStmt>(stmt, std::move(value), std::move(trueBranch), std::move(falseBranch));
+  }
+  std::unique_ptr<SymbolicCompound> symbolizeCompound(RawCompound *stmt) {
+    auto result = std::make_unique<SymbolicCompound>();
+    for (auto &s: stmt->getStmts()) {
+      result->append(symbolizeStatement(s));
+    }
+
+    return result;
+  }
+  std::unique_ptr<SymbolicWhileStmt> symbolizeWhile(RawWhileStatement* rawWhile) {
+
+    clang::WhileStmt *stmt = rawWhile->getWhileStmt();
+    llvm::outs() << "DEBUG: symbolizing while: ";
+    stmt->dump();
+    llvm::outs() << "\n";
+
+    auto symbolicWhile = std::make_unique<SymbolicWhileStmt>(stmt, boolSymbolizer.symbolize(stmt->getCond()), symbolizeCompound(rawWhile->getBody()));
+    llvm::outs() << "SymbolizedWhile: ";
+    symbolicWhile->print(llvm::outs());
+    return symbolicWhile;
+  }
+  
+  RawStatement* getParentStmt(clang::DynTypedNode node, RawStatement* raw) {
+    clang::FunctionDecl const *functionDecl = node.get<clang::FunctionDecl>();
+    if (functionDecl != nullptr) {
+      return raw;
+    }
+    clang::WhileStmt const *whileStmt = node.get<clang::WhileStmt>();
+    if (whileStmt != nullptr) {
+      llvm::outs() << "DEBUG FOUND WHILE!!!!";
+
+      //construct RawWhile if not already built
```

Comment:
memory leak?

Reply:
Memory leak of the whileStmt? I don't think so since this comes from the clang AST? 
Or am I misreading something?